c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,196 @@
+            throw new IllegalArgumentException("Exclude sensitive filter should not be null");
+        }
+
+        return new DebugOperationHandler(c2Client, bundleFilePaths, contentFilter);
+    }
+
+    @Override
+    public OperationType getOperationType() {
+        return TRANSFER;
+    }
+
+    @Override
+    public OperandType getOperandType() {
+        return DEBUG;
+    }
+
+    @Override
+    public C2OperationAck handle(C2Operation operation) {
+        String debugCallbackUrl = operation.getArgs().get(TARGET_ARG);
+        if (debugCallbackUrl == null) {
+            LOG.error("Callback URL was not found in C2 request.");
+            return operationAck(operation, operationState(NOT_APPLIED, C2_CALLBACK_URL_NOT_FOUND));
+        }
+
+        List<Path> contentFilteredFilePaths = null;
+        C2OperationState operationState;
+        try {
+            contentFilteredFilePaths = filterContent(bundleFilePaths);
+            operationState = createDebugBundle(contentFilteredFilePaths)
+                .map(bundle -> c2Client.uploadDebugBundle(debugCallbackUrl, bundle)
+                    .map(errorMessage -> operationState(NOT_APPLIED, errorMessage))
+                    .orElseGet(() -> operationState(FULLY_APPLIED, SUCCESSFUL_UPLOAD)))
+                .orElseGet(() -> operationState(NOT_APPLIED, UNABLE_TO_CREATE_BUNDLE));
+        } catch (Exception e) {
+            LOG.error("Unexpected error happened", e);
+            operationState = operationState(NOT_APPLIED, UNABLE_TO_CREATE_BUNDLE);
+        } finally {
+            ofNullable(contentFilteredFilePaths).ifPresent(this::cleanup);
+        }
+
+        LOG.debug("Returning operation ack with state {} and details {}", operationState.getState(), operationState.getDetails());
+        return operationAck(operation, operationState);
+    }
+
+    private C2OperationAck operationAck(C2Operation operation, C2OperationState state) {
+        C2OperationAck operationAck = new C2OperationAck();
+        operationAck.setOperationId(ofNullable(operation.getIdentifier()).orElse(EMPTY));
+        operationAck.setOperationState(state);
+        return operationAck;
+    }
+
+    private C2OperationState operationState(OperationState operationState, String details) {
+        C2OperationState state = new C2OperationState();
+        state.setState(operationState);
+        state.setDetails(details);
+        return state;
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try (Stream<String> fileStream = lines(path)) {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) fileStream.filter(contentFilter)::iterator);
```

Comment:
Similarly to Paths.get on the line above I think it would help readability if we would have Files.write here. I know it is a matter of taste so if you feel strongly it is ok by me.

Reply:
Removed static import